src/common/components/rangeDropdown/RangeDropdown.tsx

```diff
@@ -64,57 +64,60 @@ const RangeDropdown: React.FC<RangeDropdownProps> = ({
   const dropdown = React.useRef<HTMLDivElement | null>(null);
   const toggleButtonRef = React.useRef<HTMLButtonElement | null>(null);
 
-  //set values without validation
+  //set values with prevalidation, rest of validation on blur
   const handleInputChange = (inputType: RANGE_INPUT, val: string) => {
-    switch (inputType) {
-      case RANGE_INPUT.MIN:
-        onChange(val, maxInputValue);
-        break;
-      case RANGE_INPUT.MAX:
-        onChange(minInputValue, val);
-        break;
+    const allowChange =
+      (maxInputEndValue && val.length <= maxInputEndValue.length) ?? true;
+    if (allowChange) {
+      switch (inputType) {
+        case RANGE_INPUT.MIN:
+          onChange(val, maxInputValue);
+          break;
+        case RANGE_INPUT.MAX:
+          onChange(minInputValue, val);
+          break;
+      }
     }
   };
 
   //validate values on blur
   const handleInputBlur = (inputType: RANGE_INPUT, val: string) => {
-    const getValidatedMinValue = (val: string): string => {
+    const getValidatedValue = (val: string): string => {
```

Comment:
This method is quite hard to follow. I think that the previous version of separated `getValidatedMinValue` and `getValidatedMaxValue` was cleaner to read. Also If I understand code could be shorter something like this:
```
 const getValidatedMinValue = (val: string): string => {
return Math.Min(val, maxInputValue , maxInputEndValue)
}
 const getValidatedMaxValue = (val: string): string => {
return Math.Max(minInputStartValue, minInputValue, val)
}
```
It seems that Math functions ignore null values, but behaviour unexpectedly with NaN and undefined, so falsy values should be converted to `null` first: `const normalize(val: string) => Number(val) || null;`
I didnt test this fully so I might be also wrong as well :)

Reply:
another try:
```
const normalize = (val: string) => Number(val) || null;
const getValidatedMinValue = (val:string) => {
const values = [val, maxInputValue , maxInputEndValue].map(normalize);
return Math.min(...values);
}

```

Reply:
actually Santeri pointed out that there are a lot of code duplicates if its separated into 2 methods. So i combined it to one.

Reply:
and i agree that it might be not the best implementation, i was refactoring it couple of times, usually finding new use cases (in which order use fills fields, on load values, on search, from keyboard or typing or mouse, etc).

Reply:
if @nenonja suggestion works as expected it seems good and easily understandable, of course that can be combine into 1 function also

Reply:
As i understand the logic of his approach and what i wanted to achieve is different. His function takes 3 strings, normalizes to int values and calculates Min or Max of the set of 3. But that's not what I wanted. we have a range of allowed numbers minAllowed current maxAllowed. So for Min and Max field we want to first put the current value within the range, if its outside the range currently. So if current selected field value (works for both min and max fields) less than min => current = min. If more than Max => current = max. And then we have the third condition, and logic will differ for 2 fields. Min value should be always less than Max. We check it on blur, so depending on the selected field we will correct the values differently. So if Min is selected and its more than current Max. then we correcting Min to be the same as current Max. In opposite, when Max field is selected, and current Min value > current Max, then we will fix current Max to be equal to current Min. So the order of the filling values actually matters. As we don't want to control/change other fields value on blur, only the selected field. Hope it makes sense.

Reply:
Could we just get rid of minInputStartValue prop, same for maxInputStartValue and just use initial value given to the component. Would that simplify this little bit?

Reply:
To me @nenonja s suggestion works correctly for the most part but onyl doesn't work when it is called like this:

`handleInputBlur(RANGE_INPUT.ALL, '')`

Reply:
```
    const normalize = (val: string) => Number(val) || null;
    const getValidatedMinValue = (val: string) => {
      const values = [val, maxInputValue, maxInputEndValue]
        .map(normalize)
        .filter((n) => n) as number[];
      return Math.min(...values).toString();
    };
    const getValidatedMaxValue = (val: string) => {
      const values = [minInputStartValue, minInputValue, val]
        .map(normalize)
        .filter((n) => n) as number[];
      return Math.max(...values).toString();
    };
```

I just did little bit modifications to get it work

Reply:
I bet it could be simplified further but I don't have time now

Reply:
perhaps it needs also check that the value is not out of range:
`Math.max(0,Math.min(...values))`
`Math.min(Math.max(...values), maxInputEndValue)`